modules/imgproc/src/drawing.cpp

```diff
@@ -1879,6 +1880,45 @@ void rectangle( InputOutputArray img, Rect rec,
         rectangle( img, rec.tl(), rec.br() - Point(1<<shift,1<<shift),
                    color, thickness, lineType, shift );
 }
+void dash_rectangle(InputOutputArray _img, Point pt1, Point pt2,
+    const Scalar& color, int thickness = 1,
+    int lineType = 8, int shift = 0,int gap=5) {
+    // High thickness is going to result in a really bad look
+    thickness = 1;
+    int startX = pt1.x;
+    int endX = pt2.x;
+    int startY = pt1.y;
+    int endY = pt2.y;
+
+    while (startX + 2 < endX) {
+        Point p1(startX, pt1.y);
+        Point p2(startX + 2, pt1.y);
+        line(_img, p1, p2, color, thickness, lineType = 8, shift = 0);
```

Comment:
`lineType` should be the same as the function parameter.

Reply:
The same for `shift` I suspect.

Reply:
I have changed it now.